shim/src/handle_vcpu_kvm_get_regs.c

```diff
@@ -26,19 +26,134 @@
 
 #include <debug.h>
 #include <kvm_regs.h>
+#include <mv_rdl_entry_t.h>
+#include <mv_rdl_t.h>
+#include <mv_reg_t.h>
+#include <platform.h>
+#include <shim_vcpu_t.h>
 #include <types.h>
 
+/* Remove me */
+#define HYPERVISOR_PAGE_SIZE 13
+#define HYPERVISOR_MAX_PPS 10
+
+/* Remove me */
+/**
+ * @struct page_t
+ *
+ * <!-- description -->
+ *   @brief see /include/uapi/linux/kvm.h in Linux for more details.
+ *   @var page_t::data
+ *   Member reg holds data 
+*/
+
+struct page_t
+{
+    uint8_t data[HYPERVISOR_PAGE_SIZE];
+};
+
+/* Remove me */
+struct page_t g_shared_pages[HYPERVISOR_MAX_PPS] = {0};
+
+/* Remove me */
+uint32_t
+platform_current_cpu(void)
+{
+    return 10;
+}
+
+/* Remove me */
+void *
+shared_page_for_this_pp(void)
+{
+    uint32_t ppid = platform_current_cpu();
+    platform_expects(ppid < HYPERVISOR_MAX_PPS);
+
+    return &g_shared_pages[ppid];
+}
+
+/* Remove me */
+uint16_t
+ms_vs_op_reg_get_list(uint64_t const vsid)
+{
+    return 1;
+}
+
 /**
  * <!-- description -->
  *   @brief Handles the execution of kvm_get_regs.
  *
  * <!-- inputs/outputs -->
- *   @param ioctl_args the arguments provided by userspace
+ *   @param vcpu paramter to extract the vsid
+ *   @param regs the arguments provided by userspace
  *   @return SHIM_SUCCESS on success, SHIM_FAILURE on failure.
  */
 int64_t
-handle_vcpu_kvm_get_regs(struct kvm_regs *const ioctl_args)
+handle_vcpu_kvm_get_regs(struct shim_vcpu_t const *const vcpu, struct kvm_regs *const regs)
 {
-    (void)ioctl_args;
+    const int RAX_IDX = 0;
```

Comment:
These really should be in a header file. Probably something like `mv_to_kvm_regs_idxs.h` in the shim's main include folder. This way it can be used by the set function. Otherwise, the compiler will be forced to define these more than once in the data section of the driver which is not great.